### PR TITLE
583 years and months

### PR DESCRIPTION
**Pull request by reinhardtb**
Fixes string resolution of a timespan including years and months

### CODE REVIEW

File: src/Humanizer/TimeSpanHumanizeExtensions.cs [hunk 1]
```diff
@@ -8,190 +8,235 @@
...
-        private static int GetSpecialCaseDaysAsInteger(TimeSpan timespan, bool isTimeUnitToGetTheMaximumTimeUnit)
-        {
-            if (isTimeUnitToGetTheMaximumTimeUnit)
-            {
-                return timespan.Days;
-            }
-            return timespan.Days % _daysInAWeek;
-        }
+      return (int)(timespan.TotalDays%_daysInAYear);
+    }
```
Comment (aloisdg) [inline src/Humanizer/TimeSpanHumanizeExtensions.cs]: Also

```c#
{
    return (int)(isTimeUnitToGetTheMaximumTimeUnit
        ? Math.Floor(timespan.TotalDays / _daysInAYear)
        : timespan.TotalDays % _daysInAYear);
}
```

File: src/Humanizer/TimeSpanHumanizeExtensions.cs [hunk 2]
```diff
@@ -8,190 +8,235 @@
...
+      return amountOfTimeUnits != 0
+               ? cultureFormatter.TimeSpanHumanize(timeUnitType, Math.Abs(amountOfTimeUnits))
+               : null;
+    }
 
-            return timeParts;
-        }
+    private static List<string> CreateTimePartsWithNoTimeValue(string noTimeValue)
+    {
+      return new List<string>()
```
Comment (aloisdg) [inline src/Humanizer/TimeSpanHumanizeExtensions.cs]: You can remove the `()`

File: src/Humanizer/TimeSpanHumanizeExtensions.cs [hunk 3]
```diff
@@ -8,190 +8,235 @@
...
 
-        private static string ConcatenateTimeSpanParts(IEnumerable<string> timeSpanParts, string collectionSeparator)
-        {
-            if (collectionSeparator == null)
-            {
-                return Configurator.CollectionFormatter.Humanize(timeSpanParts);
-            }
+    private static bool IsContainingOnlyNullValue(IEnumerable<string> timeParts)
+    {
+      return (timeParts.Count(x => x != null) == 0);
```
Comment (aloisdg) [inline src/Humanizer/TimeSpanHumanizeExtensions.cs]: You can remove the `( )`

File: src/Humanizer/TimeSpanHumanizeExtensions.cs (hunk 1 above)
Comment (aloisdg) [inline src/Humanizer/TimeSpanHumanizeExtensions.cs]: You can benefit of ternary operator further 👍

File: src/Humanizer/TimeSpanHumanizeExtensions.cs [hunk 4]
```diff
@@ -8,190 +8,235 @@
 
 namespace Humanizer
 {
+  /// <summary>
```
Comment (nj) [inline src/Humanizer/TimeSpanHumanizeExtensions.cs]: Doesn't seem to obey the [`.editorconfig`](https://github.com/Humanizr/Humanizer/blob/766a10b/.editorconfig), stating 4 spaces.

This makes the diff much larger than it's supposed to.

**Comment by MehdiK** [comment]
@reinhardtb Thanks for the contribution. This has been sitting here idly for about a year now. Could you please address the comments, do a rebase and push force? Please also fix the whitespace diffs as they make it hard to review the code. Please leave us a comment. Thanks.

**Comment by MehdiK** [comment]
Actually I will close this for now. Please submit a new PR and we'll review.

**Pull request closed without merging**